### DISCUSSION TITLE
near total memory consumption on base install

### DISCUSSION THREAD

**Discussion by fsantiago07044** [mastodon/mastodon#25254] [Q&A]
hello world. my server is routinely consuming 95%+ of system RAM and even dipping into swap. periodically, resource usage drops back down to sane levels, but at some point (repeatedly) it jumps back up for sustained periods of time. about every other day or so, the server just hard locks up and must be rebooted to restore functionality. i checked journalctl around the time it last went unresponsive and the system just stopped logging cold. no messages of any kind written to log until my restart. 

my server has 2 users, me and the admin account (also me). my non-admin user account follows 83 people and has 6 followers, from various server instances. the admin account follows 1 user, me (user account). 

i haven't done much any server tuning so everything is installed at its basic default aside from what must be configured within mastodon itself. 

when i list all processes and sort by memory use, at the top is all mastodon related processes, or those of its other related components. 

i have mastodon current release (4.1.2) installed on ubuntu 20.04.6, installed from source.

my vps (SSD Nodes) is an intel 6 core, 24 GB RAM, 400 +/- GB storage. i added a 10gb swap. at the same time i tuned system swappiness to 10 and cache pressure to 50. i wasn't sure about the swap, but since i had plenty of storage i figured it'd be fine. 

all media is stored on s3 (Wasabi), with an nginx proxy in front of it with mastodon pointed at the nginx hosted s3 alias host (alias hosted on same server as mastodon). 

any ideas on where to begin looking? Thanks.

**Comment by renchap** [comment]
Can you check which Mastodon processes are using your RAM?

I expect to see some Sidekiq processes, as we seem to have a known memory leak in a (yet unidentified) job.

**Reply by fsantiago07044** [reply]
Sorry for the narrow browser display. Laying in bed doing this on my phone....
![image](https://github.com/mastodon/mastodon/assets/8866211/a65f266b-bea4-4536-ae8d-bad8b6878f56)

**Reply by fsantiago07044** [reply]
It's currently quiet / ok. Swap does have around ~250mb used atm; All in single digit percentages.

**Comment by fsantiago07044** [comment]
i'm scratching my head. it hasn't gone down again since in a hard lock. i'm trying to retrace my steps, as this server was a new setup on a new VPS hosting provider that i transposed from a prior one. maybe i screwed up early on and have since "fixed" it. not sure yet, but still examining. what i can say, as i monitor the system, is RAM use periodically spikes as i said to around 95% +/- and then comes back down some time later. and swap remains lightly used. i'm still looking and trying to keep track of logs, resources...

**Reply by renchap** [reply]
You might want to install [Netdata](https://www.netdata.cloud), this is a simple way to tracking resources on a single server. It can track per-process resources as well.

**Reply by fsantiago07044** [reply]
yes agreed and i had already done so.

**Comment by fsantiago07044** [comment]
sorry. if you saw my last comment (now deleted), i commented on the wrong discussion.

**Comment by fsantiago07044** [comment]
i spoke too soon; the server is hard down again (locked). vps hosting provider shows it as being in an on / running state, but it's completely unresponsive. also can't ssh into it. now, sadly, i can't gain remote console access w/o a restart (the host SSD Nodes requires restarts to enable console access). so if i do that it'll be back online again anyway. in my system monitoring (Netdata), the logs show it stopped responding at around 2:45 pm ET, earlier today. as of 2:27 pm, ram usage was at 94%. at 2:45, it reported ram use at 43%. and then it went dead. 

any thoughts?

i'm nervous about this being attributable to the vps provider, ssd nodes. i've ran vps' on digital ocean (droplets) and linode with years long up time; never a loss of availability unless i directly caused it, or the provider had some sort of connectivity issue. but the vps itself always remained up. this server has gone down every few days since i created it. this mastodon server instance existed on DO before moving over to ssd nodes and never a hiccup. same os, same method of install, and it last had the same version of mastodon installed as here now. 

i know this is not a topic for here, so i left it at the bottom, last. but does anyone here know of ssd nodes and their reputation? does anyone recommend anyone else? i moved for the better pricing. but maybe you get what you pay for? (like restarts to enable console. i only ever use console for troubleshooting. and a restart will likely destroy any evidence of what you're investigating. that's nuts. DO had an always available console thru your customer account portal. ssd nodes also warns you to disable console when you're done, for security. another restart. when you enable it, they give you an ip, port, & password, to a vnc session of the console. i'm not impressed.). 

anyway, enough for now. i guess i'll restart it since i have basically no other options. maybe i'll just leave the console enabled at least. but i suspect it's hosted on the vps itself so a hard lock will take that down too i'd bet.

**Comment by fsantiago07044** [comment]
ok check this out. i'm watching the console now and i see this:
![Screenshot 2023-06-08 at 11 58 20 PM](https://github.com/mastodon/mastodon/assets/8866211/8784b369-a10e-4265-8b33-fc135f8210e7)

those watchdog messages look suspect. and i can't ssh in yet. sshd is listed there. hmmm...

**Comment by fsantiago07044** [comment]
eventually ssh let me in, a WHILE later.

**Comment by fsantiago07044** [comment]
i'm reading now some bad reviews of the host. i guess maybe i chose poorly perhaps. DO never let me down, but were so much pricier. at this point i'm willing to bet it's them and not mastodon. where else can i find cheaper (than DO) vps hosting?

**Reply by renchap** [reply]
We are using Hetzner Cloud and it works fine: https://www.hetzner.com/cloud

**Comment by fsantiago07044** [comment]
Cool. I’ll try them. Their prices are a tad higher than ssd but still a
fraction of DO. And I can host in VA, us. Thanks.

On Fri, Jun 9, 2023 at 2:50 AM Renaud Chaput ***@***.***>
wrote:

> We are using Hetzner Cloud and it works fine:
> https://www.hetzner.com/cloud
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mastodon/mastodon/discussions/25254#discussioncomment-6129584>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACDUTIYZJJTETJCVMWLW65LXKLBSXANCNFSM6AAAAAAYYVM2VU>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
--

Fabian A. Santiago

OpenPGP:
0x643082042dc83e6d94b86c405e3daa18a1c22d8f

**Comment by fsantiago07044** [comment]
ok, all migrated now. I'll wait and see. thanks for your help.

**Comment by fsantiago07044** [comment]
i can confirm it's fine now on a new host. sorry for the trouble.
